source/scheme/mus_scheme_derived_quantities_type_module.f90

```diff
@@ -192,78 +193,83 @@ function mus_assign_derived_functions_ptr(label_stencil, label_fluid) result(get
   getQuantities%kineticEnergy_from_vel_dens_ptr => null()
   getQuantities%rho0Inv_ptr => null()
 
-  if (trim(label_fluid) == 'fluid') then
+  select case (trim(scheme_kind))
+  case('fluid')
     getQuantities%pdfEq_iDir_ptr => get_pdfEq_compressible_iDir
     getQuantities%momentum_from_vel_dens_ptr => get_momentum_from_vel_dens_compressible
     getQuantities%kineticEnergy_from_vel_dens_ptr => get_kineticEnergy_from_vel_dens_compressible
     getQuantities%rho0Inv_ptr => get_rho0Inv_compressible
-  else if (trim(label_fluid) == 'fluid_incompressible') then
+  case('fluid_incompressible')
     getQuantities%pdfEq_iDir_ptr => get_pdfEq_incompressible_iDir
     getQuantities%momentum_from_vel_dens_ptr => get_momentum_from_vel_dens_incompressible
     getQuantities%kineticEnergy_from_vel_dens_ptr => get_kineticEnergy_from_vel_dens_incompressible
     getQuantities%rho0Inv_ptr => get_rho0Inv_incompressible
-  else 
-    write(*,*) 'fluid type = "', trim(label_fluid), '"'
-    call tem_abort("get_pdfEq_iDir not set for fluid type")
-  end if
+  case default
+    write(logUnit(1),*) 'fluid type = "', trim(scheme_kind), '"'
+    write(logUnit(1),*) "Warning: get_pdfEq_iDir not set for fluid type"
```

Comment:
Use proper write formating instead of *.
Example:
 ```
write(logUnit(1), "(A)") 'fluid type = "' // trim(scheme_kind) // '"'
write(logUnit(1), "(A)") "Warning: get_pdfEq_iDir not set for fluid type"
```